File: android/src/main/java/com/evotor/modules/UserModule.java
package com.evotor.modules;

import com.facebook.react.bridge.Callback;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.evotor.utilities.Writer;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import ru.evotor.framework.users.Grant;
import ru.evotor.framework.users.User;
import ru.evotor.framework.users.UserApi;

/**
 * Created by a.lunkov on 24.10.2017.
 */

public class UserModule extends ReactContextBaseJavaModule {

    private final ReactApplicationContext context;

    public UserModule(ReactApplicationContext c) {
        super(c);
        context = c;
    }

    @Override
    public String getName() {
        return "UserModule";
    }

    @Override
    public Map<String, Object> getConstants() {
        return new HashMap<>();
    }

    @ReactMethod
    public void getAllUsers(Callback callback) {
        List<User> users = UserApi.getAllUsers(context);
        callback.invoke(users == null ? null : Writer.INSTANCE.writeUsers(users));
    }

    @ReactMethod
    public void getAuthenticatedUser(Callback callback) {
        User user = UserApi.getAuthenticatedUser(context);
        callback.invoke(user == null ? null : Writer.INSTANCE.writeUser(user));
    }

    @ReactMethod
    public void getAllGrants(Callback callback) {
        List<Grant> grants = UserApi.getAllGrants(context);
        callback.invoke(grants == null ? null : Writer.INSTANCE.writeGrants(grants));
    }

    @ReactMethod
    public void getGrantsOfAuthenticatedUser(Callback callback) {
        List<Grant> grants  = UserApi.getGrantsOfAuthenticatedUser(context);
        callback.invoke(grants == null ? null : Writer.INSTANCE.writeGrants(grants));
    }

}

File: android/src/main/java/com/evotor/modules/UserModule.kt
package com.evotor.modules

import com.evotor.converter.tojs.UserWriter
import com.facebook.react.bridge.Callback
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

import java.util.HashMap

import ru.evotor.framework.users.Grant
import ru.evotor.framework.users.User
import ru.evotor.framework.users.UserApi

/**
 * Created by a.lunkov on 24.10.2017.
 */

class UserModule(private val context: ReactApplicationContext) : ReactContextBaseJavaModule(context) {

    override fun getName(): String {
        return "UserModule"
    }

    override fun getConstants(): Map<String, Any>? {
        return HashMap()
    }

    @ReactMethod
    fun getAllUsers(callback: Callback) {
        callback.invoke(UserApi.getAllUsers(context)?.let { UserWriter.writeUsers(it) })
    }

    @ReactMethod
    fun getAuthenticatedUser(callback: Callback) {
        callback.invoke(UserApi.getAuthenticatedUser(context)?.let { UserWriter.writeUser(it) })
    }

    @ReactMethod
    fun getAllGrants(callback: Callback) {
        callback.invoke(UserApi.getAllGrants(context)?.let { UserWriter.writeGrants(it) })
    }

    @ReactMethod
    fun getGrantsOfAuthenticatedUser(callback: Callback) {
        val grants = UserApi.getGrantsOfAuthenticatedUser(context)
        callback.invoke(UserApi.getGrantsOfAuthenticatedUser(context)?.let { UserWriter.writeGrants(it) })
    }

}
